planner/core/logical_plan_builder.go

```diff
@@ -1559,14 +1574,37 @@ func (b *PlanBuilder) resolveGbyExprs(p LogicalPlan, gby *ast.GroupByClause, fie
 		schema: p.Schema(),
 	}
 	for _, item := range gby.Items {
+		hasParam := false
 		resolver.inExpr = false
-		retExpr, _ := item.Expr.Accept(resolver)
+		var retExpr ast.Node
+		switch x := item.Expr.(type) {
+		case *driver.ParamMarkerExpr:
+			hasParam = true
+			if b.ctx.GetSessionVars().InPrepare {
+				continue
+			}
+			newItem, isNull, err := expression.ParamToByItemNode(b.ctx, x)
+			if err != nil {
+				err := errors.Errorf("Unknown column '%+v' in 'group statement'", "?")
```

Comment:
ditto

Reply:
The updated PR should address the comment.